app/src/preHome.js

```diff
@@ -0,0 +1,201 @@
+      event.preventDefault();
+      return listSessions();
+    }
+
+    const handleCreateSession = event => {
+      event.preventDefault();
+    }
+
+    async function listSessions() {
+        const apiName = 'MTADemoAPI';
+        const path = 'listSessions';
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: {
+            PatientId: searchVal,
+          },
+        };
+        const result =  await API.get(apiName, path, myInit);
+        setSessions(result.data);
+        return result;
+    }
+
+    async function listPatients() {
+        const apiName = 'MTADemoAPI';
+        const path = 'listPatients';
+        const parameters = (searchVal === '') ? {} : {PatientId: searchVal} 
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: parameters
+        };
+        const result =  await API.get(apiName, path, myInit); 
+        return result;
+    }
+
+    async function listHealthCareProfessionals() {
+        const apiName = 'MTADemoAPI';
+        const path = 'listHealthCareProfessionals';
+        const parameters = (searchVal === '') ? {} : {HealthCareProfessionalId: searchVal} 
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: parameters
+        };
+    
+        const result =  await API.get(apiName, path, myInit); 
+        return result;
+    }
+ 
+    async function createPatient() {
+        const apiName = 'MTADemoAPI';
+        const path = 'createPatient';
+        const myInit = { 
+            //   headers: { 
+            //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+            //   },
+            response: true,
+            queryStringParameters: {
+                'PatientName': searchVal,
+            },
+        };
+    
+        const result =  await API.post(apiName, path, myInit); 
+        return result;
+    }
+
+    async function createHealthCareProfessional() {
+        const apiName = 'MTADemoAPI';
+        const path = 'createHealthCareProfessional';
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getIdToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: {
+            'HealthCareProfessionalName': searchVal,
+          },
+        };
+    
+        const result =  await API.post(apiName, path, myInit); 
+
+        return result;
+    }
+
+    const pBack = () => {
```

Comment:
Could you change the function names to bit more descriptive names? Like P=Patient so the function can be named onPatientBack() or something similar.

Reply:
Changed. https://github.com/aws-samples/medical-transcription-analysis/blob/eafae50489bbda5b2cd751d3679ef7a7ae8d11ce/app/src/preHome.js#L116